lib/puppet/provider/file_line/ruby.rb

```diff
@@ -43,7 +52,7 @@ def handle_create_with_match()
 
     File.open(resource[:path], 'w') do |fh|
       lines.each do |l|
-        fh.puts(regex.match(l) ? resource[:line] : l)
+        fh.puts(match_regex.match(l) ? resource[:line] : l)
```

Comment:
Yeah, the handle_create_with_match function will not be used to write the line unless resource[:match] is not nil. So the creation of the regex that match_regex returns should not fail in this block.